src/test/java/com/google/api/gax/grpc/ApiCallableTest.java

```diff
@@ -171,11 +172,16 @@ public void retryKeepFailing() {
         .thenReturn(
             Futures.<Integer>immediateFailedFuture(
                 Status.UNAVAILABLE.withDescription("foobar").asException()));
+    FakeNanoClock clock = new FakeNanoClock(System.nanoTime());
     ApiCallable<Integer, Integer> callable =
         ApiCallable.<Integer, Integer>create(callInt)
             .retryableOn(retryable)
-            .retrying(testRetryParams, EXECUTOR);
-    callable.call(1);
+            .retrying(testRetryParams, EXECUTOR, clock);
+    ListenableFuture<Integer> future = callable.futureCall(1);
+    long advanceTimeBy =
+        TimeUnit.MILLISECONDS.toNanos(testRetryParams.getTotalTimeout().getMillis() + 1);
+    clock.setCurrentNanoTime(System.nanoTime() + advanceTimeBy);
```

Comment:
We need to make the clock thread-safe since we're accessing it from multiple threads.